### ISSUE TITLE
Question about the cpp code

### ISSUE THREAD

**Issue by Tsukumo1** [noirritchandra/Lamb#2]
hi thank u for your sharing! I have a little question about the cpp code. When I try to run the rmarkdown, it shows something wrong in the cpp file line 436, DOUBLE_EPS was not declared.
may I check what is "DOUBLE_EPS" refer to?

**Comment by noirritchandra** [comment]
Thanks for your interest in my work. In my compiler, DOUBLE_EPS works fine
but this is an issue people complained about before. Can you try replacing
the DOUBLE_EPS by datum::eps? datum::eps  is in the Armadillo library:

[image: image.png]



Let me know if this works.


Best,


Noirrit

**Comment by Tsukumo1** [comment]
Hi! Thank you very much for your reply. That's working! I have another question about the cpp code, why don't you return parameters in the DL_mixture? Which variables are used to represent parameters......and I also want to ask about the data preprocessing method you used in PART 6 of your paper. 

Sorry for all the questions... ! But if it is convenient, could you also provide the code in the paper?

**Comment by noirritchandra** [comment]
>
> *Hi! Thank you very much for your reply. That's working! I have another
> question about the cpp code, why don't you return parameters in the
> DL_mixture? Which variables are used to represent parameters...... - * Since
> our primary interest is in clustering high-dimensional data, we are only
> returning the cluster allocation variables acrosss MCMC samples. The
> residual error variance parameters σj2's and the factor loading matrix Λ
> are the only parameters you can return; it would be a simple change in the
> return statement in the DL_mixture function.
>
You can also consider the mixture parameters [image: \mu_{i}]'s and [image:
\Delta_{i}]'s in eqn (8) of the arxived paper as parameters but due to
label switching those are not identifiable. Hence we are not returning
these parameters.


> *and I also want to ask about the data preprocessing method you used in
> PART 6 of your paper.*- Let me know your questions, I'll try my best to
> answer those.
>
> *Sorry for all the questions... ! But if it is convenient, could you also
> provide the code in the paper?*- We do not have plans for that since the
> codes are available on Github.
>

I hope this helps.

Best

**Comment by noirritchandra** [comment]
>
> *Hi! Thank you very much for your reply. That's working! I have another
> question about the cpp code, why don't you return parameters in the
> DL_mixture? Which variables are used to represent parameters...... - * Since
> our primary interest is in clustering high-dimensional data, we are only
> returning the cluster allocation variables acrosss MCMC samples. The
> residual error variance parameters σj2's and the factor loading matrix Λ are
> the only parameters you can return; it would be a simple change in the
> return statement in the DL_mixture function.
>
You can also consider the mixture parameters [image: \mu_{i}]'s and [image:
\Delta_{i}]'s in eqn (8) of the arxived paper as parameters but due to
label switching those are not identifiable. Hence we are not returning
these parameters.


> *and I also want to ask about the data preprocessing method you used in
> PART 6 of your paper.*- Let me know your questions, I'll try my best to
> answer those.
>
> *Sorry for all the questions... ! But if it is convenient, could you also
> provide the code in the paper?*- We do not have plans for that since the
> codes are available on Github.
>

I hope this helps.

Best
Noirrit

**Comment by Tsukumo1** [comment]
Hi for PART 6 of your paper, `Cells with low read counts are discarded, as we lack reliable gene expression measurements for these cells, and data are normalized following Lun et al.(2016). We remove non-informative genes using M3Drop (Andrews and Hemberg,2018).`. I tried the method you mentioned, but I want to know exactly how you got a final dataset with n = 531 cells and p = 7,666 genes. Thank u for your reply!

**Comment by Tsukumo1** [comment]
Hi I also want to check if you choose TMM or DESeq to normalize the data? I got a bad cluster by using both of them. So I want to confirm what's wrong with it. Looking for your reply!

**Comment by noirritchandra** [comment]
Hi,

I apologize for not responding to your earlier email. I can not really
provide you with the exact script; I was part of a bioinformatics lab where
I used their scripts to preprocess the data. However, I do not have access
to those anymore.

Since the data is really high-dimensional with not that many cells, I
discarded those genes with <30% observed counts across all cells.

I hope this helps.

Best,

Noirrit

**Comment by Tsukumo1** [comment]
Thank u for your reply! I'll try then. And another question, did u modify the parameters in LAMB when dealing with the real data?

**Comment by noirritchandra** [comment]
No, I did not. Those are exactly the same as suggested in the manuscript.

Noirrit
